#! /bin/sh /usr/share/dpatch/dpatch-run
## 18_vdr-1.3.26-pagedown-fix.dpatch by Klaus Schmidinger <Klaus.Schmidinger@cadsoft.de>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix handling of the "page down" function

@DPATCH@
diff -urNad vdr-1.3.26/osdbase.c /tmp/dpep.Devsn7/vdr-1.3.26/osdbase.c
--- vdr-1.3.26/osdbase.c	2005-06-12 12:44:22.000000000 +0200
+++ /tmp/dpep.Devsn7/vdr-1.3.26/osdbase.c	2005-06-12 17:35:50.779644016 +0200
@@ -362,7 +362,7 @@
   if (current >= 0) {
      if (current < first)
         first = current;
-     else if (current - first >= displayMenuItems)
+     if (current - first >= displayMenuItems || current == last)
         first = current - displayMenuItems + 1;
      }
   if (current != oldCurrent || first != oldFirst) {
